packages/plugins/rpc-manager/src/lib/index.ts

```diff
@@ -20,25 +22,66 @@ export default class RpcManager {
   private events: EmbarkEvents;
   public _accounts: any[] | null = null;
   public _nodeAccounts: any[] | null = null;
+  private blockchainConfig: EmbarkConfig["blockchainConfig"];
   constructor(private readonly embark: Embark) {
     this.events = embark.events;
     this.logger = embark.logger;
     this.rpcModifierEvents = new Events() as EmbarkEvents;
+    this.blockchainConfig = this.embark.config.blockchainConfig;
     this.init();
   }
 
-  private async init() {
+  protected get web3() {
+    return (async () => {
+      if (!this._web3) {
+        await this.events.request2("blockchain:started");
+        let provider;
+        if (this.embark.currentContext.includes(constants.contexts.test)) {
+          provider = await this.events.request2("blockchain:client:vmProvider");
```

Comment:
Why check for that here? Shouldn't the `blockchain:client:provider` request be the only source for providers and be it that checks what type of provider it should return? Especially since it doesn't **just** depends on the conext. The tests can still connect to a node using `--node`

Reply:
This is due to the way we handle passing around the VM provider in tests. If we instead spun up ganache-cli normally for tests, and exposed the endpoints as we would expect, we wouldn't have to handle these kinds of cases.

To be more clear, the reason we need to do this is because we are connecting **directly to the node**, instead of connecting to the proxy. In 99% of other modules, we connect to the proxy (even during tests) by way of `blockchain:client:provider`. However, during tests, we are using the VM provider which does not expose connection endpoints and forces us to pass the provider around to instantiate web3. Because of this, we have to request the VM provider using `blockchain:client:vmProvider` and instantiate web3 using the returned provider.

Very good point regarding `--node`. Do you know of a way we could conditionally test for this instead of `if (this.embark.currentContext.includes(constants.contexts.test))`?

Reply:
> Shouldn't the blockchain:client:provider request be the only source for providers and be it that checks what type of provider it should return? Especially since it doesn't just depends on the conext.

I agree, I had a similar thought a few times, however, that would mean that during testing modules would connect directly to the node instead of connecting to the proxy. My opinion is that keeping it this way lets us be very specific around what we want to do in the module. Moreover, I feel like we should change the tests to **not** use the VM provider, so that we don't need to keep adding special cases for it. If we were to do so, then we could get rid of `blockchain:client:vmProvider` altogether and rely on `blockchain:client:provider` (with optional endpoint parameter) throughout the codebase.

Reply:
Added additional handling for the case when `--node=embark`.